ui/appui-react/src/appui-react/statusfields/MessageCenter.tsx

```diff
@@ -5,227 +5,165 @@
+          const iconSpec = MessageManager.getIconSpecFromDetails(details);
+          const message: MessageType = details.briefMessage;
+          /* istanbul ignore else */
+          if ((tab === "error" && isProblemStatus(details)) || tab === "all") {
+            return (
+              <MessageCenterMessage
+                key={index.toString()}
+                message={message}
+                details={details.detailedMessage}
+                icon={<Icon iconSpec={iconSpec} className={iconClassName} />}
+              />
+            );
+          }
+          return <></>;
+        } else {
+          return <></>;
+        }
+      });
+  };
 
-        tabRows.push(
-          <MessageCenterMessage
-            key={index.toString()}
-            icon={<Icon iconSpec={iconSpec} className={iconClassName} />}
+  const tabs = (
+    <>
+      {["all", "error"].map((tabType) => (
+        <Tabs.Panel value={tabType} key={tabType}>
+          <div
+            className={"nz-footer-messageCenter-dialog"}
+            style={{ overflowY: "scroll" }}
           >
-            <MessageRenderer message={message} useSpan />
-            {details.detailedMessage && (
-              <Text variant="small">
-                <MessageRenderer message={details.detailedMessage} />
-              </Text>
+            {messages.length < 0 && (
+              <span className="nz-message-prompt">No Messages. </span>
             )}
-          </MessageCenterMessage>
-        );
+            {messages.length > 0 && getMessages(tabType)}
+          </div>
+        </Tabs.Panel>
+      ))}
+    </>
+  );
+
+  return (
+    <Popover
+      content={
+        <>
+          <TitleBar title={title}></TitleBar>
+
+          <Tabs.Wrapper type="pill" role="tablist">
+            <Tabs.TabList>
+              <Tabs.Tab label="All" key="all" value="all" />
+              <Tabs.Tab label="Error" key="error" value="error" />
+            </Tabs.TabList>
+            {tabs}
+          </Tabs.Wrapper>
+        </>
       }
-    });
-
-    return tabRows;
-  }
-
-  private isProblemStatus(priority: OutputMessagePriority): boolean {
-    // See priority values in DgnPlatform defined in NotificationManager
-
-    if (
-      priority === OutputMessagePriority.Error ||
-      priority === OutputMessagePriority.Fatal
-    )
-      return true;
-
-    return false;
-  }
-
-  private setIsOpen(isOpen: boolean) {
-    this.setState({ isOpen });
-  }
-}
+      applyBackground
+    >
+      <div style={divStyle} title={`${messages.length} ${title}`}>
+        <Button
+          styleType="borderless"
+          startIcon={
+            !notify ? (
+              <SvgChat />
+            ) : (
+              <NotificationMarker status={notifyStatus()}>
+                <SvgChat />
+              </NotificationMarker>
+            )
+          }
+        >
+          <div // eslint-disable-line jsx-a11y/click-events-have-key-events
```

Comment:
Why is this div needed? Seems like we can re-factor to use buttons props, i.e. `Button["startIcon"]`

Reply:
Much better now, I think we can still remove the `div` and probably even `span` for the label?